### DISCUSSION TITLE
PHP OpenSSL

### DISCUSSION THREAD

**Discussion by brentonv** [getgrav/grav#3601] [Help]
Can someone please explain how can I install without PHP OpenSSL for offline testing as I can't pass the install checklist

`"PHP OpenSSL (Secure Sockets Library) is required but not installed"`

**Comment by spamhater-grav** [comment]
This is part grav and part php and the symfony-cli to create certificates , 

Could you tell us what operating system you are using for more specific details I will be focusing based on a windows system

In your php install, you need to ensure that you have openssl available to your php install
This will mostly be installed to your **EXT** directory of your php install 

You then need to enable it in the php.ini by nu-commenting the line
`extension=openssl `

If you download the symfony-cli from github  place this in either your grav directory or your php directory, or in its own separate directory as long as it gets added to the path.


run the command 
`symfony server:ca:install` 

This will create in .symfony5\cert directory , a **root-ca.pem** and a personal certificate **default.p12** usually in a .symfony5 directory,
You need to install these to your certificate server, and also good to install via your browser certificate store.


Back to the `php .ini`

Now search for these sections 


[curl]
; A default value for the CURLOPT_CAINFO option. This is required to be an
; absolute path.
; you would add above a full absolute path to the certificate folder created by symfony
`curl.cainfo = "C:\Users\nickc\.symfony5\certs\rootCA.pem"`


[openssl]
; The location of a Certificate Authority (CA) file on the local filesystem
; to use when verifying the identity of SSL/TLS peers. Most users should
; not specify a value for this directive as PHP will attempt to use the
; OS-managed cert stores in its absence. If specified, this value may still
; be overridden on a per-stream basis via the "cafile" SSL stream context
; option.
;again absolute path to the cafile
`openssl.cafile= "C:\Users\nickc\.symfony5\certs\rootCA.pem"`

; If openssl.cafile is not specified or if the CA file is not found, the
; directory pointed to by openssl.capath is searched for a suitable
; certificate. This value must be a correctly hashed certificate directory.
; Most users should not specify a value for this directive as PHP will
; attempt to use the OS-managed cert stores in its absence. If specified,
; this value may still be overridden on a per-stream basis via the "capath"
; SSL stream context option.

; this is just to the directory holding those certificate
`openssl.capath= "C:\Users\nickc\.symfony5\certs\"`

this will now give you php openssl connectivity, if you need to add to a webserver, you will need to add it to a virtual host. (more info needed)
